File: commons/src/main/java/com/msx7/android/commons/MConfig.java
package com.msx7.android.commons;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

import okhttp3.Interceptor;

/**
 * 所 属 包：com.lovehome.android.aijiazhusou.common.flux
 * 文 件 名：MConfig
 * 描    述：
 * 作    者：xiaowei
 * 时    间：2018/5/8
 */
public class MConfig {
    String            baseUrl         = "https://www.github.com";
    List<Interceptor> interceptors    = new LinkedList();
    List<Interceptor> netInterceptors = new LinkedList();


    public MConfig() {
    }

    public MConfig(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public MConfig(String baseUrl, Interceptor... interceptor) {
        this(baseUrl);
        this.interceptors.addAll(Arrays.asList(interceptor));
    }

    public MConfig setBaseUrl(String baseUrl) {
        this.baseUrl = baseUrl;
        return this;
    }

    public MConfig addInterceptor(Interceptor... interceptor) {
        this.interceptors.addAll(Arrays.asList(interceptor));
        return this;
    }

    public MConfig addNetInterceptor(Interceptor... interceptor) {
        this.netInterceptors.addAll(Arrays.asList(interceptor));
        return this;
    }
}

File: commons/src/main/java/com/msx7/android/commons/MConfig.kt
package com.msx7.android.commons

import okhttp3.Interceptor

class MConfig {
    constructor()
    constructor(baseUrl: String) {
        this.baseUrl = baseUrl
    }

    constructor(baseUrl: String, vararg interceptor: Interceptor) {
        this.baseUrl = baseUrl;
        this.interceptor = this.interceptor.plus(interceptor)
    }

    var baseUrl: String = "https://www.github.com"
    var interceptor: Array<Interceptor> = arrayOf()
    var netInterceptor: Array<Interceptor> = arrayOf()

    fun baseUrl(url: String): MConfig {
        this.baseUrl = url
        return this
    }

    fun addInterceptor(interceptor: Interceptor): MConfig {
        this.interceptor = this.interceptor.plus(interceptor)
        return this
    }

    fun addInterceptor(vararg interceptor: Interceptor): MConfig {
        this.interceptor = this.interceptor.plus(interceptor)
        return this
    }

    fun addNetInterceptor(vararg interceptor: Interceptor): MConfig {
        this.netInterceptor = netInterceptor.plus(interceptor)
        return this
    }

    fun addNetInterceptor(interceptor: Interceptor): MConfig {
        this.netInterceptor = netInterceptor.plus(interceptor)
        return this
    }

}